collect/collect.go

```diff
@@ -256,16 +296,26 @@ func (c *InventoryCollection) Upload() error {
 	}
 	defer res.Body.Close()
 
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		log.Error().Err(err).Int("status", res.StatusCode).Msg("reading response")
+		return errors.Wrap(err, "reading response")
+	}
+
 	if res.StatusCode != http.StatusOK && res.StatusCode != http.StatusCreated {
-		body, err := io.ReadAll(res.Body)
-		if err != nil {
-			log.Error().Err(err).Int("status", res.StatusCode).Msg("reading response")
-			return errors.Wrap(err, "reading response")
-		}
 		log.Error().Int("status", res.StatusCode).Str("body", string(body)).Msg("")
 		return errors.New("upload failed")
 	}
 
+	metaDataResponse := &uploadResponse{}
+	if err := json.Unmarshal(body, metaDataResponse); err != nil {
+		return errors.Wrap(err, "unmarshal response")
+	}
+	t := time.Now()
+	c.MetaData.Updated = &t
```

Comment:
You might consider whether you want to protect access to the `MetaData` data with a mutex (https://pkg.go.dev/sync) there might be a small chance of getting inconsistent data?